### ISSUE TITLE
LibraryNodes Written to Scene

### ISSUE THREAD

**Issue by bryantcurto** [KhronosGroup/COLLADA2GLTF#236]
`node`s stored within a COLLADA file's `library_nodes` are being converted to GLTF and written to the GLTF scene, as can be seen [here](https://github.com/KhronosGroup/COLLADA2GLTF/blob/master/src/COLLADA2GLTFWriter.cpp#L505-L509).
Considering that elements within all other `library_*` elements are not used without being referenced, I believe that this is an error.

Looking the code over, here are my thoughts on a solution:
- Separate translation of a `node` from COLLADA to GLTF from the task of adding a GLTF `node` into a group. In this way, `node`s can be handled as needed after translating based off of context. For example, `node`s in the COLLADA `visual_scene` can be added directly to the GLTF scene after translation. Meanwhile, `node`s in `library_nodes` can be [registered](https://github.com/KhronosGroup/COLLADA2GLTF/blob/master/src/COLLADA2GLTFWriter.cpp#L431), so that it may be instanced/cloned later.
- Keep things as they are and just store the library nodes to a dummy group. For example:
```
bool COLLADA2GLTF::Writer::writeLibraryNodes(const COLLADAFW::LibraryNodes* libraryNodes) {
	std::vector<GLTF::Node*> dummy_group;
	return this->writeNodesToGroup(&dummy_group, libraryNodes->getNodes());
}
```

Let me know if there's anything I'm missing.

**Comment by lasalvavida** [comment]
[GLTFAsset::removeUnusedNodes](https://github.com/KhronosGroup/COLLADA2GLTF/blob/master/GLTF/src/GLTFAsset.cpp#L686) already prunes out any nodes that aren't in the final scene graph, so this definitely isn't necessary.

If there's a compelling optimization argument to be made here, I'd be open to it, but it seems like it probably will just complicate conversion unnecessarily.

**Comment by bryantcurto** [comment]
I believe that there has been a misunderstanding, so here's a list of what I observed:
- I generated a GLTF file from my COLLADA file, which contains `instance_node` and `library_nodes` elements
- Viewing the GLTF file, I observed
  - my model being displayed correctly
  - a set of meshes, which are instanced in my model, but floating next to my model
- I edited my source COLLADA file, adding a `node` (containing an `instance_geometry`) to the set of `library_nodes`, but without creating any `instance_node` referencing the `node` in any `visual_scene`.
- I passed this COLLADA file through `COLLADA2GLTF`
- Viewing the results, I saw:
  - my model being displayed correctly
  - the same set of meshes floating next to my model
  - an instance of the geometry that I added in the `library_nodes` floating next to my model
- I concluded that the `node`s within `library_nodes` were being added to the GLTF scene, in addition to the `instance_node`s referencing them.
- Looking within the `COLLADA2GLTF` source, I found the [following line](https://github.com/KhronosGroup/COLLADA2GLTF/blob/master/src/COLLADA2GLTFWriter.cpp#L505-L509), which appears to be adding the library nodes to the scene:
```
bool COLLADA2GLTF::Writer::writeLibraryNodes(const COLLADAFW::LibraryNodes* libraryNodes) {
	GLTF::Asset* asset = this->_asset;
	GLTF::Scene* scene = asset->getDefaultScene();
	return this->writeNodesToGroup(&scene->nodes, libraryNodes->getNodes());
}
```
- Changing the definition of this function to the following fixed the issue:
```
	std::vector<GLTF::Node*> dummy_group;
	return this->writeNodesToGroup(&dummy_group, libraryNodes->getNodes());
```

**Comment by lasalvavida** [comment]
Ah, gotcha. Thanks @bryantcurto. Unfortunately, this is a little more complicated to fix than just writing the nodes to an empty group because of memory ownership. They will need to be added to a group and then removed as they are instantiated. The nodes left in the group (not attached to the asset) should be freed when conversion finishes.

**Comment by lasalvavida** [comment]
If you're interested in submitting a pull request, that would be great, otherwise I'll get to it when I have time.

**Comment by Febfire** [comment]
@lasalvavida Does this bug fixed?
I got the same problem.

**Comment by lasalvavida** [comment]
Hi @febfire, it will be fixed when #239 merges

**Comment by lasalvavida** [comment]
Fixed in #239
